src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -246,5 +246,75 @@ void Add(INamedTypeSymbol symbol)
 
             return builder?.ToImmutable() ?? ImmutableArray<INamedTypeSymbol>.Empty;
         }
+
+        private bool IsMethodNotImplemented(OperationBlockAnalysisContext context)
+        {
+            // Check to see if the method just throws a NotImplementedException
+            // Mark Members as static should not trigger for methods containing only NotImplementedException 
+            
+            // Note that VB method bodies with 1 action have 3 operations.
+            // The first is the actual operation, the second is a label statement, and the third is a return
+            // statement. The last two are implicit in these scenarios.
+                        
+            var operationBlocks = context.OperationBlocks;
+            if (operationBlocks.Any(operation => operation.IsOperationNoneRoot()))
+            {
+                operationBlocks = operationBlocks.Where(operation => !operation.IsOperationNoneRoot()).ToImmutableArray();
```

Comment:
This could be a `WhereAsArray` call.

https://github.com/dotnet/roslyn-analyzers/blob/57318576d6a39f4858e1110b611ce201505183ef/src/Utilities/Extensions/IEnumerableExtensions.cs#L74

Reply:
And by that I mean, it handles the `Any` case without double iteration and allocations.

Reply:
Thanks for review, updated the same, also in unused parameters file where same construct is used.